src/floating.tsx

```diff
@@ -45,14 +47,15 @@ type Context = {
     };
     refs: Pick<ReturnType<typeof useFloating>["refs"], "setReference" | "setFloating"> & {
         arrowRef: React.MutableRefObject<HTMLDivElement | null>;
+        listRef: React.MutableRefObject<HTMLElement[] | null>;
     };
     getReferenceProps: ReturnType<typeof useInteractions>["getReferenceProps"];
     getFloatingProps: ReturnType<typeof useInteractions>["getFloatingProps"];
 };
 
 const FloatingContext = React.createContext<Context | null>(null);
 
-const useFloatingContext = () => {
+export const useFloatingContext = () => {
```

Comment:
The floating context is implementation detail for this whole stuff, so I'd prefer not to just export it. I thought about alternatives and I think the most useful one is to offer another hook (that is exported):

```tsx
export const useFloatingItemProps = () => {
    const context = useFloatingContext();
    return (index: number) => ({
        tabIndex: context.activeIndex === index ? 0 : -1,
        ref: (node: HTMLElement) => {
            if (context.refs.listRef.current !== null) {
                context.refs.listRef.current[i] = node;
            }
        },
        ...context.getItemProps(),
    });
};
```

Then in using components it would like this:

```tsx
const itemProps = useFloatingItemProps();
...
<ul>{items.map((item, i) => <li {...itemProps(i)} />)}</ul>
```

This would expose very little about the floating internals and requires little code in the using component. I have not tried this yet though!

Oh and you might notice the `getItemProps` above. In [the docs](https://floating-ui.com/docs/useListNavigation) it seems like one should forward the `getItemProps` (returned by `useInteractions`) to the items.

Reply:
> In [the docs](https://floating-ui.com/docs/useListNavigation) it seems like one should forward the getItemProps (returned by useInteractions) to the items.

Ooh, good catch. This even seems to fix another issue that I experienced when doing https://github.com/elan-ev/tobira/pull/1001. Thank you for taking another look at the docs, sorry I missed this!